### DISCUSSION TITLE
If no files are in the cache, does it mean that ccache isn't working?

### DISCUSSION THREAD

**Discussion by Eldadc** [ccache/ccache#255] [Support]
I running eclipse CLI command that run ICC compiler.
ccache eclipse .......
does ccache support eclipse ?
Thanks

**Comment by afbjorklund** [comment]
No files is a bad sign... Also check `ccache -s` so that they are not just rejected...

Normally you would configure your *compiler* inside the IDE, to use ccache.

**Comment by Eldadc** [comment]
Thanks for the quick reply.
So ccache can intercept eclipse cli command ? I run a script with eclipse commands.
Is it enough to invoke icc and icpc from the shell?

**Comment by afbjorklund** [comment]
I believe so, but have not checked those compilers - ccache supports gcc and clang.

**Comment by Eldadc** [comment]
Thanks so much on your info.
Is it on the roadmap?
Is it possbile to make a sanity test to check icc support.....
Thanks

**Comment by jrosdahl** [comment]
ccache seemed to work with the Intel C++ compiler back in year 2010, see 97b2778196df8afa05f88c7f5fadb9d3e7954aeb. But ICC support isn't tested when making new ccache releases, so the current state is unknown, but it's likely working.

It's not on the road map to add tests for it, but I would welcome contributions.
